src/ConformalTracking.cc

```diff
@@ -685,6 +685,31 @@ void ConformalTracking::processEvent(LCEvent* evt) {
 
     streamlog_out(DEBUG9) << " Fit direction " << ((conformalTrack->m_kalmanFitForward) ? "forward" : "backward")
                           << std::endl;
+    streamlog_out(DEBUG9) << " Track hits after fit = " << track->getTrackerHits().size() << std::endl;
+
+    // If the track is too short (usually less than 7 hits correspond to a vertex track)
+    // the fit is tried using the inverted direction
+    if (track->getTrackerHits().size() < 7 || track->getChi2() == 0.) {
```

Comment:
```suggestion
    if (track->getTrackerHits().size() < 7 || fitError != MarlinTrk::IMarlinTrack::success) {
```
* Not 100 percent sure about the namespaces, or if another include is needed
* The 7 should become a parameter at some point.